python/ray/actor.py

```diff
@@ -740,12 +743,16 @@ def __setstate__(self, state):
 
 
 def make_actor(cls, num_cpus, num_gpus, resources, actor_method_cpus,
-               checkpoint_interval):
+               checkpoint_interval, max_reconstructions):
     if checkpoint_interval is None:
         checkpoint_interval = -1
+    if max_reconstructions is None:
+        max_reconstructions = 0
 
     if checkpoint_interval == 0:
         raise Exception("checkpoint_interval must be greater than 0.")
+    if max_reconstructions < 0 or max_reconstructions > 2**32 - 1:
```

Comment:
yep, it should be 2**31 - 1. 
however, to avoid potential overflow, I'll just define the max value constant as 2**30.
Using -1 is also fine, but it requires some special checks in the code. Thus, I prefer to use a large number to represent infinite reconstruction.

Reply:
Sounds good!